File: android/src/main/java/com/iconshot/detonator/request/InputSetValueRequest.java
package com.iconshot.detonator.request;

import android.widget.EditText;

import com.iconshot.detonator.Detonator;
import com.iconshot.detonator.renderer.Edge;

public class InputSetValueRequest extends Request<String> {
    public InputSetValueRequest(Detonator detonator, IncomingRequest incomingRequest) {
        super(detonator, incomingRequest);
    }

    @Override
    public void run() {
        String value = decodeData(String.class);

        Edge edge = getComponentEdge();

        Edge elementEdge = edge.children.get(0);

        EditText view = (EditText) elementEdge.element.view;

        view.setText(value);
    }
}

File: ios/Sources/Detonator/Request/InputSetValueRequest.swift
import UIKit

class InputSetValueRequest: Request {
    override public func run() {
        let value: String! = decodeData()
        
        let edge = getComponentEdge()!
        
        let elementEdge = edge.children[0]
        
        let element = elementEdge.element! as! InputElement
        
        element.updateValue(value: value)
        
        end()
    }
}
